push_to_docker.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# The image build step occurs only when a build is triggered by a commit being merged into the "master" branch.
+# Given that the "master" branch is protected, the only way to trigger a build from the "master" branch is when a PR
+# is merged into it. This way, we avoid storing images from non-stable branches.
+VERSION=$(cat ./version/version.go | grep "var VERSION" | awk ' { print $4 } ' | sed s/\"//g)
+VERSION_REGEX="^$VERSION$"
+TAG=$(curl https://registry.hub.docker.com/v1/repositories/tfgco/khan/tags | \
```

Comment:
Could we use a more specialized command to check this? `docker pull` would definitely work, but I suspect we can avoid the needless pull. `docker search`, maybe?

Reply:
I'm not satisfied with this code either, seems too complex for such a simple task 👍.

Unfortunately `docker search` won't work. As the [documentation](https://docs.docker.com/engine/reference/commandline/search/) states it only works with the filters `stars`, `is-automated` and `is-official`.

As for `docker pull` it will work and its much simpler to understand, with the tradeoff of (possibly) downloading the image in case it already exists. Gonna keep looking for a better alternative and update the PR as soon as I find one.

Reply:
Improved this in c795574506168613b740b37ae400657b3cf4c1c5